packages/mattermost-redux/src/reducers/entities/channels.ts

```diff
@@ -336,6 +335,10 @@ export function myMembers(state: RelationOneToOne<Channel, ChannelMembership> =
         } = action.data;
         const member = state[channelId];
 
+        if (amount === amountRoot) {
+            return state;
+        }
```

Comment:
AFAIU this check will return true for root posts messages and false for replies in threads, don't we need to increment counts on both of these occasions?

Reply:
Yeah, I'd expect to see this look more like `amount === 0 && amountRoot === 0` since that's the only time this would return a new object with the same values

Reply:
changed